### ISSUE TITLE
error CS0539: 'LinkXmlInjector.OnBeforeRun(BuildReport, UnityLinkerBuildPipelineData)' in explicit interface declaration is not found among members of the interface that can be implemented

### ISSUE THREAD

**Issue by haliik** [LastAbyss/SimpleGraphQL-For-Unity#34]
Thank you for the wonderful library.

Unity 2021.2.0b13 (macOS)

`error CS0539: 'LinkXmlInjector.OnBeforeRun(BuildReport, UnityLinkerBuildPipelineData)' in explicit interface declaration is not found among members of the interface that can be implemented`

/Editor/LinkXmlInjector.cs
```
        void IUnityLinkerProcessor.OnBeforeRun(BuildReport report, UnityLinkerBuildPipelineData data)
        {
        }

        void IUnityLinkerProcessor.OnAfterRun(BuildReport report, UnityLinkerBuildPipelineData data)
        {
        }
```

[https://unity3d.com/jp/unity/whats-new/2021.1.0](https://unity3d.com/jp/unity/whats-new/2021.1.0)
- Scripting: Deprecated: Deprecated IUnityLinkerProcessor.OnAfterRun.
- Scripting: Deprecated: Deprecated IUnityLinkerProcessor.OnBeforeRun.

**Comment by johanhelsing** [comment]
That's unfortunate.

I put a post on the unity forums about what the recommended replacement is: https://forum.unity.com/threads/the-current-state-of-link-xml-in-packages.995848/#post-7532342

**Comment by johanhelsing** [comment]
Oh. sorry I misread. It's just the two methods we don't need that are gone, not the actual interface. We could probably just put them behind an #ifdef then.

**Comment by haliik** [comment]
Thank you for your quick reply.

**Comment by skjalgsm** [comment]
Now that Unity 2021.2 is out it would be nice to surround these in a ifdef and have this issue resolved

**Comment by NavidK0** [comment]
> Now that Unity 2021.2 is out it would be nice to surround these in a ifdef and have this issue resolved

I'll take a look as soon as I get a chance. There's been a number of API changes in 2021.2 that require updating a bunch of assets.

**Comment by NavidK0** [comment]
Should be fixed with c5e7634394fa498768a74f7e687b0ad3a7f67f7f.
